service-registry/src/main/java/org/apache/servicecomb/serviceregistry/version/Version.java

```diff
@@ -74,34 +78,52 @@ private short parseNumber(String name, String allVersion, String version) {
     return value;
   }
 
-  public Version(short major, short minor, short patch) {
+  public Version(short major, short minor, short patch, short build) {
     this.major = major;
     this.minor = minor;
     this.patch = patch;
+    this.build = build;
     this.version = combineStringVersion();
     this.numberVersion = combineVersion();
   }
 
   private String combineStringVersion() {
-    return major + "." + minor + "." + patch;
+    StringBuilder stringBuilder = new StringBuilder();
+    stringBuilder.append(major)
+        .append(".")
+        .append(minor)
+        .append(".")
+        .append(patch)
+        .append(".")
+        .append(build);
+    return stringBuilder.toString();
   }
 
+  // 1.0.0 equals 1.0.0.0
   private long combineVersion() {
-    return (long) major << 32 | (long) minor << 16 | (long) patch;
+    return (long) major << 48 | (long) minor << 32 | (long) patch << 16 | (long) build;
   }
 
+  @VisibleForTesting
```

Comment:
why VisibleForTesting?

Reply:
it's only used in Junit

Reply:
it's a public class, and have normal getter/setter?
why only can be used in junit?

Reply:
ok